### PR TITLE
Feature linux support

### PR DESCRIPTION
**Pull request by CaptainDario**
This PR should add linux support.

Environment:
* Ubuntu 22.04

Changes:
* Load TF Lite binary
* update examples (same as macos) to support linux

Notes:
* Camera is disabled for the examples on desktop as this is not really supported

Progress:
- [x] load TF Lite on linux
- [x] test examples
  - [x] audio classification
    - uses some platform channels that do not work on linux  
  - [x] image classification mobilenet
    - only from file works
  - [x] live object detection ssd mobile net
    - uses camera package that only works on iOS / mobile 
  - [x] object detection ssd mobile net
    - only from file works
  - [x] style transfer
    - works 
  - [x] super resolution esrgan
    - works
  - [x] text classification
    - works

### CODE REVIEW

**Comment by CaptainDario** [comment]
@PaulTR I tested linux support for the same examples as MacOS (quite a few new one since the Mac PR). Somehow those PRs added a lot of problems with dart analyze for me locally, not sure what to do with those.

**Comment by PaulTR** [comment]
Awesome, thanks! I'll see what I can do to get that passing - I need to head into the office and get a linux machine up and running with a Flutter dev environment, so it might be a hot minute unfortunately (doing a bit of work travel this week), but I'll try to get that sorted out soon.

**Comment by CaptainDario** [comment]
No worries, let me know once you get to it. Maybe I will already add Windows support in the mean time.

**Comment by PaulTR** [comment]
Alright so I have Flutter set up with vs code on a Debian machine. Are there any special steps needed to get one of the samples working? I started up digit classifier (since it doesn't need a camera) and getting some errors around cmake. If this isn't something you've run into before though, let me know and I'll see if I can dig deeper into it. Thanks!

ptruiz@ptruiz-p620:~/Documents/code/flutter-tflite/example/digit_classification$ flutter run
Launching lib/main.dart on Linux in debug mode...
CMake Error at flutter/generated_plugins.cmake:22 (add_subdirectory):
  add_subdirectory given source
  "flutter/ephemeral/.plugin_symlinks/tflite_flutter/linux" which is not an
  existing directory.
Call Stack (most recent call first):
  CMakeLists.txt:92 (include)


Building Linux application...                                           
Exception: Unable to generate build files

**Comment by CaptainDario** [comment]
Hey, thanks for checking this out!
It seems like you are trying the digit classifier, I didn't add support to this example. I checked all the examples I checked on MacOS.
See the Initial PR message for the ones I checked.

**Comment by PaulTR** [comment]
So I'm seeing that same error with object detection and text classification FWIW :)

**Comment by CaptainDario** [comment]
Well, the only setup that I did with all the examples was to add the tf lite binary to the `blobs` folder. Otherwise, it should work. Maybe it's related to my 'outdated' version of Flutter.

**Comment by PaulTR** [comment]
@st-duymai Do you happen to have a Linux machine? I don't think I'll be back at the office before leave for doing this review. If not, I can ask someone internally, too. Thanks!

**Comment by st-duymai** [comment]
> @st-duymai Do you happen to have a Linux machine? I don't think I'll be back at the office before leave for doing this review. If not, I can ask someone internally, too. Thanks!

I don't currently have a Linux machine, but I can set one up to test it.

@CaptainDario Can you help me resolve the conflict?

**Comment by CaptainDario** [comment]
Yeah, sure! Do you encounter the same issue as PaulTR?

**Comment by CaptainDario** [comment]
@PaulTR I think I found the problem that you were encountering and I pushed a fix. @st-duymai Let me know if you have problems setting this up.

**Comment by PaulTR** [comment]
Sounds good, I'll see if I can make my way in to an office tomorrow or Wednesday to give it one more shot. Thanks!

**Comment by st-duymai** [comment]
@CaptainDario I frequently encounter this problem when building Linux applications. Have you ever dealt with this issue before?
`/home/ubuntu/Documents/flutter-tflite/example/super_resolution_esrgan/build/linux/x64/debug/bundle/super_resolution_esrgan: symbol lookup error: /home/ubuntu/Documents/flutter-tflite/example/super_resolution_esrgan/build/linux/x64/debug/bundle/blobs/libtensorflowlite_c-linux.so: undefined symbol: _ZN6tflite3ops7builtin12Register_NEGEv`

**Comment by CaptainDario** [comment]
No I have not seen this issue, how did you build the binary?

The one I used can be found [here](https://github.com/CaptainDario/DaKanji-Dependencies/releases/download/v3.2.0/libtensorflowlite_c-linux.so)

**Comment by st-duymai** [comment]
Thanks @CaptainDario I will check again with your .so library. I may have built it wrong at some step.

**Comment by st-duymai** [comment]
Hi @PaulTR ,

After check with .so library that @CaptainDario  provide above, I can confirm that the following five examples are working properly:

1. Style transfer
2. Text classification
3. Super resolution esrgan
4. Object detection SSD Mobilenet(only file work)
5. Image classification

However, two examples are not working:
1. Audio classification
2. Live object detection SSD Mobilenet

**Comment by st-duymai** [comment]
Hi @CaptainDario 
Here is my libtensorflow_cc.so build step. Is this step correct?
```
sudo apt-get install cmake
git clone https://github.com/tensorflow/tensorflow.git tensorflow_src
mkdir tflite_build
cd tflite_build
cmake ../tensorflow_src/tensorflow/lite/c
cmake --build . -j

**Comment by CaptainDario** [comment]
> Hi @CaptainDario Here is my libtensorflow_cc.so build step. Is this step correct?
> 
> ```
> sudo apt-get install cmake
> git clone https://github.com/tensorflow/tensorflow.git tensorflow_src
> mkdir tflite_build
> cd tflite_build
> cmake ../tensorflow_src/tensorflow/lite/c
> cmake --build . -j
> ```

You forgot to checkout a specific release, I guess the state you are currently in is broken. I used r2.12

**Comment by PaulTR** [comment]
> Hi @PaulTR ,
> 
> After check with .so library that @CaptainDario provide above, I can confirm that the following five examples are working properly:
> 
> 1. Style transfer
> 2. Text classification
> 3. Super resolution esrgan
> 4. Object detection SSD Mobilenet(only file work)
> 5. Image classification
> 
> However, two examples are not working:
> 
> 1. Audio classification
> 2. Live object detection SSD Mobilenet

Awesome, thanks for checking on those. Yeah those two are expected to not work with desktop, so that's fine. I think once we have those conflicts resolved we should be pretty golden then for all three desktop versions! :)

**Comment by PaulTR** [comment]
Screenshot from 2023-11-21 07-58-45
Just confirmed image classification on my work machine with my own .so file, so we're looking good on my side too

**Comment by st-duymai** [comment]
Hi @CaptainDario Would you mind resolving the conflict in this PR? Thank you.

**Comment by CaptainDario** [comment]
Yeah sure! I did not notice that there are conflicts.

**Comment by CaptainDario** [comment]
done

File: example/text_classification/README.md [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 
 This basic Text Classification app demonstrates the usage of tflite_flutter_plugin.
 
 ![DEMO GIF](demo.gif)
+
+## Setup
+
+Run `sh scripts/download_model.sh` to download models.
+
+## Supported platforms
```
Comment (st-duymai) [inline example/text_classification/README.md:17]: "Supported platforms" overlap. I think we should remove this.

**Comment by st-duymai** [comment]
Hi @CaptainDario . I think this PR look greate and ready to merge. I just have one minor request regarding the readme file, which I mentioned earlier. Thank you for your  hard work.

File: example/text_classification/README.md (hunk 1 above)
Comment (st-duymai) [reply example/text_classification/README.md]: I have resolved this issue. Thanks.

Comment (CaptainDario) [reply example/text_classification/README.md]: Yes you are right, thank you for resolving this!

**Comment by CaptainDario** [comment]
@st-duymai are you able to release new tf lite version on pub?

**Comment by st-duymai** [comment]
@CaptainDario The new version was published a few days ago (tflite_flutter: ^0.10.4). Please check.

**Comment by CaptainDario** [comment]
@st-duymai thanks for the heads up, I tried it!

**Pull request merged**